src/common/CvtFormat.cpp

```diff
@@ -0,0 +1,1678 @@
+	{
+		switch (pattern)
+		{
+			case Format::Y:
+			case Format::YY:
+			case Format::YYY:
+			case Format::YYYY:
+			case Format::YEAR:
+			case Format::RR:
+			case Format::RRRR:
+			case Format::Q:
+			case Format::MM:
+			case Format::MON:
+			case Format::MONTH:
+			case Format::RM:
+			case Format::WW:
+			case Format::W:
+			case Format::D:
+			case Format::DAY:
+			case Format::DD:
+			case Format::DDD:
+			case Format::DY:
+			case Format::J:
+				return !desc->isTime();
+
+			case Format::HH:
+			case Format::HH12:
+			case Format::HH24:
+			case Format::MI:
+			case Format::SS:
+			case Format::SSSSS:
+			case Format::FF1:
+			case Format::FF2:
+			case Format::FF3:
+			case Format::FF4:
+			case Format::FF5:
+			case Format::FF6:
+			case Format::FF7:
+			case Format::FF8:
+			case Format::FF9:
+			case Format::AM:
+			case Format::PM:
+				return !desc->isDate();
+
+			case Format::TZH:
+			case Format::TZM:
+			case Format::TZR:
+				return desc->isDateTimeTz();
+
+			default:
+				fb_assert(false);
+				return false;
+		}
+	}
+
+	constexpr UCHAR expectDatetimeToDscType(EXPECT_DATETIME expectedDatetime)
+	{
+		switch (expectedDatetime)
+		{
+			case expect_timestamp:
+				return dtype_timestamp;
+			case expect_timestamp_tz:
+				return dtype_timestamp_tz;
+			case expect_sql_date:
+				return dtype_sql_date;
+			case expect_sql_time:
+				return dtype_sql_time;
+			case expect_sql_time_tz:
+				return dtype_sql_time_tz;
+			default:
+				fb_assert(false);
+				return 0;
+		}
+	}
+
+	// Start reading format and comparing it to patterns, until we stop finding similarities
+	template <FB_SIZE_T PatternsSize>
+	constexpr std::string_view getPatternFromFormat(const char* format, const char* const (&patterns)[PatternsSize],
+		FB_SIZE_T formatLength, FB_SIZE_T formatStart, FB_SIZE_T& offset)
+	{
+		std::string_view patternStr;
+		for (; offset < formatLength; offset++)
+		{
+			patternStr = std::string_view(format + formatStart, offset - formatStart + 1);
+			bool isFound = false;
+
+			for (int j = 0; j < PatternsSize; j++)
+			{
+				if (!strncmp(patterns[j], patternStr.data(), patternStr.length()))
+				{
+					isFound = true;
+					break;
+				}
+			}
+			if (!isFound)
+				break;
+		}
+
+		// Our current pattern contains real pattern + one extra symbol, so we need to drop it
+		// if it's not a end of format
+		if (offset != formatLength)
+			return patternStr.substr(0, patternStr.length() - 1);
+		return patternStr;
+	}
+
+	void invalidPatternException(std::string_view pattern, Callbacks* cb)
+	{
+		cb->err(Arg::Gds(isc_invalid_date_format) << string(pattern.data(), pattern.length()));
+	}
+
+	void incompatibleDateFormatException(std::string_view pattern, Callbacks* cb)
+	{
+		cb->err(Arg::Gds(isc_incompatible_date_format_with_current_date_type) << string(pattern.data(), pattern.length()));
+	}
+
+//---------------------------------------------------------------------------------------------------------------------
```

Comment:
Please indent comments.

Reply:
I'm not quiet get it, you mean `// ----`?